server/mdm/apple/mobileconfig/mobileconfig.go

```diff
@@ -18,8 +18,25 @@ const (
 	// FleetdConfigPayloadIdentifier is the value for the PayloadIdentifier used
 	// by fleetd to read configuration values from the system.
 	FleetdConfigPayloadIdentifier = "com.fleetdm.fleetd.config"
+
+	// FleetdConfigProfileName is the value for the PayloadDisplayName used by
+	// fleetd to read configuration values from the system.
+	FleetdConfigProfileName = "Fleetd configuration"
+
+	// FleetdFileVaultProfileName is the value for the PayloadDisplayName used
+	// by Fleet to configure FileVault and FileVault Escrow.
+	FleetFileVaultProfileName = "Disk encryption"
 )
 
+// FleetReservedProfileNames returns a map of PayloadDisplayName strings
+// that are reserved by Fleet.
+func FleetReservedProfileNames() map[string]struct{} {
+	return map[string]struct{}{
+		FleetdConfigProfileName:   {},
+		FleetFileVaultProfileName: {},
+	}
```

Comment:
> creating a new Windows profile named "Disk encryption" must also fail, because names are unique across platforms so if it exists as a custom Windows profile name, then the Apple profile cannot be created

Are we enforcing uniqueness of all profile names across platforms or are you suggesting that only the Fleet reserved names would be unique per platform? My understanding is that we only enforce uniqueness of profile names at the table-level and we have separate tables for Windows and Apple. For example, it would be possible for Team A to have profiles named "Foo" for Windows and Apple

Reply:
> Are we enforcing uniqueness of all profile names across platforms

Yes, the UNIQUE index takes care of uniqueness per-platform, but when creating profiles we do enforce that it is unique across platforms, e.g. here for Windows (and similar for macos): https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/microsoft_mdm.go#L1329-L1335

> For example, it would be possible for Team A to have profiles named "Foo" for Windows and Apple

This is not possible with the INSERT-level verification that we do.

Reply:
Ah, thank you! I wasn't aware that the INSERT-level verification was added.